core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -49,8 +51,9 @@ class FastAppend extends SnapshotProducer<AppendFiles> implements AppendFiles {
   private final List<DataFile> newFiles = Lists.newArrayList();
   private final List<ManifestFile> appendManifests = Lists.newArrayList();
   private final List<ManifestFile> rewrittenAppendManifests = Lists.newArrayList();
-  private ManifestFile newManifest = null;
+  private List<ManifestFile> cachedNewManifests = null;
```

Comment:
@RussellSpitzer Do you mean the `caching` part here? Here just rename the variable because it is plural now.